src/modules/jcms_rest/jcms_rest.module

```diff
@@ -0,0 +1,30 @@
+<?php
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_entity_presave().
+ * 
+ * Preprocess output fields and save in field_processed_json on
+ * node to improve API performance.
+ * 
+ * @param EntityInterface $entity
+ */
+function jcms_rest_entity_presave(EntityInterface $entity) {
+  if ($entity->getEntityType()->id() == 'node'){
+    if ($entity->hasField('field_content') && $entity->hasField('field_processed_json')) {
+      $pluginManager = \Drupal::service('plugin.manager.rest');
+      $processed = $pluginManager->createInstance('blog_article_item_rest_resource', [])->processFieldContent($entity->get('field_content'));
```

Comment:
this line seems hardcoded to `blog article`. Is it intentional?

Reply:
if so, the `if()` condition can be made more clear?

Reply:
The problem here is accessing a class method from a hook. To do this a class needs to be instantiated that uses AbstractRestResourceBase as its base class but it does not matter which one so blog_article_item_rest_resource was chosen. An alternative would be to make processFieldContent() a static method but this would have wider implications.

Reply:
Seems to use references to `$this` so I agree on the wider surgery required. Extract a `any_abstract_rest_resource()` function that instantiates this, with this comment as a docblock to communicate this?